src/components/RuleCard/index.jsx

```diff
@@ -751,6 +772,8 @@ RuleCard.propTypes = {
   onRuleDelete: func,
   // If true, the card will hide all buttons.
   readOnly: bool,
+  onSignoff: func.isRequired,
+  onRevoke: func.isRequired,
```

Comment:
`onSignoff` and `onRevoke` should not be required when `readOnly` is set to true. I would remove the `isRequired` and add a comment saying that those properties are required if `readOnly` is false.

Reply:
Good point!